### PR TITLE
[FIX] web: handle early option selections in autocomplete

### PR DESCRIPTION
**Pull request by tdecaluwe**
The autocomplete component intercepts certain key presses allowing the end-user to select an option from the autocomplete dropdown using either the enter or tab keys. Before this commit those keypresses where only intercepted when the autcomplete selection list was already completely loaded.

Consider now for example an x2many list where the lines can be selected using the autocomplete component. To add a line to the list the end-user will usually type part of the record name and then select it with tab or enter. Three things can happen:

* The enter keypress is handled after the list of options is fully loaded
* The enter keypress is handled while waiting for the list of options to load
* The enter keypress is handled while still debouncing the input, no request to load the options has been made so far

The last two cases commonly occur when inputting lines with a barcode scanner for example. In this case the enter keypress is ignored by the autocomplete component and is instead handled by the list renderer component. This generally results in the list renderer trying to create a new line but failing because of missing required fields in the previous line. This seems more or less acceptable. However, if required fields are already filled out or not present in the list view, a new line is created while the current one remains empty.

This behavior is clearly counterintuitive, as such, this commit proposes to wait for handling tab and enter keypresses while debouncing the input or loading the list of options.

opw-3235288

### CODE REVIEW

File: addons/web/static/tests/views/list_view_tests.js [hunk 1]
```diff
@@ -17256,4 +17256,48 @@ QUnit.module("Views", (hooks) => {
...
+                </field>
+            </form>`,
+            mockRPC: async (route, args, performRPC) => {
+                if (args.method === "name_search") {
+                    response = performRPC(route, args);
+                    return deferred;
+                } else {
+                    return performRPC(route, args);
+                }
+            },
```
Comment (mcm-odoo) [inline addons/web/static/tests/views/list_view_tests.js:17286]:
```diff
            mockRPC: async (route, { method }) => {
                if (method === "name_search") {
                    await deferred;
                }
            },
```

File: addons/web/static/tests/views/list_view_tests.js [hunk 2]
```diff
@@ -17256,4 +17256,48 @@ QUnit.module("Views", (hooks) => {
...
+
+        await triggerEvent(document.querySelector('.o_field_x2many_list_row_add a'), null, "click");
+
+        const input = document.activeElement;
+        input.value = 'alu';
+        triggerEvent(document.activeElement, null, "input"),
+        triggerHotkey("Enter"),
+        await nextTick();
+
+        deferred.resolve(await response);
```
Comment (mcm-odoo) [inline addons/web/static/tests/views/list_view_tests.js:17297]: `deferred.resolve();`

**Comment by mcm-odoo** [comment]
@robodoo r+

**Comment by mcm-odoo** [comment]
@robodoo rebase-ff

**Comment by bodedra** [comment]
This PR break the autocomplete text typing on the relational fields.

I submitted support ticket: #3372451

**Pull request closed without merging**